source/common/stats/thread_local_store.cc

```diff
@@ -106,6 +109,18 @@ void ThreadLocalStoreImpl::shutdownThreading() {
   shutting_down_ = true;
 }
 
+void ThreadLocalStoreImpl::RegisterMutexContentionGauge() {
+  mutex_contention_instances = &default_scope_->counter("mutex.contention.instances");
+  mutex_contention_wait_cycles = &default_scope_->gauge("mutex.contention.wait_cycles");
+
+  absl::RegisterMutexTracer([](const char* msg, const void* obj, int64_t wait_cycles) -> void {
+    UNREFERENCED_PARAMETER(msg);
+    UNREFERENCED_PARAMETER(obj);
+    mutex_contention_instances->inc();
+    mutex_contention_wait_cycles->set(wait_cycles);
```

Comment:
wouldn't we want to capture the total number of wait cycles? i.e. leave this as a counter rather than a gauge?

Reply:
I rewrote this to some extent in 2f9191d, but feel free to comment there on specifically which counters/gauges we ought to store.

Reply:
i am not sure we want to dup these in stats,  maybe just provide admin access to the atomics?